code/game/objects/items/weapons/chemical_flamethrower/chemical_flamethrower.dm

```diff
@@ -0,0 +1,287 @@
+		return FALSE
+
+	var/length = length(canisters)
+	var/difference = amount
+	for(var/i in 0 to length)
+		var/obj/item/chemical_canister/canister = canisters[length - i]
+		if(canister.ammo - difference <= 0)
+			difference -= canister.ammo
+			canister.ammo = 0
+		else
+			canister.ammo -= difference
+			difference = 0
+
+		if(!difference)
+			break
+
+	if(difference)
+		return FALSE
+	return TRUE
+
+/obj/item/chemical_flamethrower/extended
+	name = "extended capacity chemical flamethrower"
+	desc = "A flamethrower that accepts two chemical canisters to create lasting fires."
+	max_canisters = 2
+
+/obj/item/chemical_flamethrower/extended/nuclear
+	name = "\improper Syndicate extended capacity chemical flamethrower"
+	desc = "A flamethrower that accepts two chemical canisters to create lasting fires. As black as the ash of your enemies."
+	syndicate = TRUE
+
+/obj/item/chemical_flamethrower/extended/nuclear/Initialize(mapload)
+	. = ..()
+	for(var/i in 1 to max_canisters)
+		canisters += new /obj/item/chemical_canister/extended/nuclear
+	update_canister_stats()
+
+/obj/item/chemical_canister
+	name = "chemical canister"
+	desc = "A simple canister of fuel. Does not accept any pyrotechnics except for welding fuel."
+	icon = 'icons/obj/chemical_flamethrower.dmi'
+	icon_state = "normal"
+	container_type = OPENCONTAINER
+	/// How much ammo do we have? Empty at 0.
+	var/ammo = 100
+	/// Which reagent IDs do we accept
+	var/list/accepted_chemicals = list("fuel")
+	/// The burn temperature of our currently stored chemical
+	var/chem_burn_temp = T0C + 300
+	/// The burn duration of our currently stored chemical
+	var/chem_burn_duration = 10 SECONDS
+	/// How many firestacks will our reagent apply
+	var/fire_applications = 1
+	/// The currently stored reagent ID
+	var/current_reagent_id
+	/// How many units of the reagent do we need to have it's effects kick in?
+	var/required_volume = 10
+	/// Do we have a locked in reagent type?
+	var/has_filled_reagent = FALSE
+	/// Are we silent on the first change of reagents?
+	var/first_time_silent = FALSE // The reason for this is so we can have canisters that spawn with reagents but don't announce it on `Initialize()`
+
+/obj/item/chemical_canister/Initialize(mapload)
+	. = ..()
+	create_reagents(50)
+
+/obj/item/chemical_canister/on_reagent_change()
+	if(has_filled_reagent)
+		visible_message("<span class='notice'>[src] doesn't accept any new chemicals!</span>")
+		return
+
+	if(!reagents.get_master_reagent_id() || !(reagents.get_master_reagent_id() in accepted_chemicals))
+		reagents.clear_reagents()
+		visible_message("<span class='notice'>[src] doesn't accept the most present chemical!</span>")
+		return
+
+	current_reagent_id = reagents.get_master_reagent_id()
+	reagents.isolate_reagent(current_reagent_id)
+	var/has_enough_reagents = reagents.total_volume >= required_volume
+
+	if(!first_time_silent)
+		visible_message("<span class='notice'>[src] removes all reagents except for [current_reagent_id]. \
```

Comment:
I feel like this should probably be a to_chat, or if not it should in some way indicated as the device saying this. This feels like a *lot* of information to just dump onto the user at once.

Reply:
It was meant to be the device saying it yeah
I'm fine with either thing, what do you think?

Reply:
I think you might want to switch to `audible_message`, or maybe just preface the message with something like `[src]'s speaker beeps:`, like what we do with the rsg/defib. You could also atom say, but I think that might be a bit much for runechat haha

Reply:
It is done